frontend/src/main/pages/Moderate.js

```diff
@@ -14,13 +23,44 @@ const Moderate = () => {
     return <Navigate to="/" />;
   }
 
+  const handleApprove = async (alias) => {
+    try {
+      await axios.put("/api/currentUser/updateAliasModeration", null, {
+        params: { id: alias.id, approved: true },
+      });
+      toast.success(
+        `Alias "${alias.proposedAlias}" for ID ${alias.id} approved!`,
+      );
+    } catch (err) {
+      toast.error(`Error approving alias: ${err.message || "Unknown error"}`);
+    }
+  };
+
+  const handleReject = async (alias) => {
+    try {
+      await axios.put("/api/currentUser/updateAliasModeration", null, {
+        params: { id: alias.id, approved: false },
+      });
+      toast.success(
+        `Alias "${alias.proposedAlias}" for ID ${alias.id} rejected!`,
+      );
+    } catch (err) {
+      toast.error(`Error rejecting alias: ${err.message || "Unknown error"}`);
+    }
+  };
+
   return (
     <BasicLayout id="moderate-page" data-testid="moderate-page">
       <div className="pt-2">
         <h1>Moderation Page</h1>
         <p>
           This page is accessible only to admins and moderators. (Placeholder)
```

Comment:
It's no longer a placeholder, is it?

```suggestion
          This page is accessible only to admins and moderators. 
```